sklearn/metrics/tests/test_pairwise.py

```diff
@@ -192,18 +193,18 @@ def test_pairwise_boolean_distance(metric):
         pairwise_distances(X.astype(bool), Y=Y, metric=metric)
 
     # Check that no warning is raised if X is already boolean and Y is None:
-    with pytest.warns(None) as records:
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", UserWarning)
         pairwise_distances(X.astype(bool), metric=metric)
-    assert not [w.message for w in records]
 
 
 def test_no_data_conversion_warning():
     # No warnings issued if metric is not a boolean distance function
     rng = np.random.RandomState(0)
     X = rng.randn(5, 4)
-    with pytest.warns(None) as records:
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", UserWarning)
```

Comment:
From the commit that introduced the test: https://github.com/scikit-learn/scikit-learn/commit/4b9e12e73b52382937029d29759976c3ef4aee3c

```suggestion
        warnings.simplefilter("error", DataConversionWarning)
```